packages/migrate-config/src/migrate-config.js

```diff
@@ -675,6 +680,25 @@ function migrateConfigObject(migration, config) {
 		});
 
 		if (needsCompat) {
+			/*
+			 * When even one `extends` item needs compat, we need to mark every
+			 * plugin as needing compat. This is because the `fixupConfigRules`
+			 * function will be called on the entire object, and if any of those
+			 * plugins is also referenced in `plugins`, the user will get a
+			 * "can't redefine plugin" error.
+			 */
+			extendsArray.forEach(extend => {
+				if (extend.startsWith("plugin:")) {
+					const pluginName = extend.slice(7);
```

Comment:
```suggestion
					const pluginName = extend.slice(7, extend.indexOf("/"));
```

In `pluginsNeedingCompatCache`, we need to store the plugin name without `/config-name` for this to work.